src/dosbox.cpp

```diff
@@ -467,40 +468,49 @@ void DOSBOX_SetMachineTypeFromConfig(Section_prop* section)
 		                         arguments->machine);
 	}
 
-	std::string mtype = section->Get_string("machine");
-	svgaCard = SVGA_None;
-	machine = MCH_VGA;
-	int10.vesa_nolfb = false;
+	const auto machine_str = section->Get_string("machine");
+
+	svga_type = SvgaType::None;
+	machine   = MachineType::Vga;
+
+	int10.vesa_nolfb  = false;
 	int10.vesa_oldvbe = false;
-	if      (mtype == "cga")      { machine = MCH_CGA; mono_cga = false; }
-	else if (mtype == "cga_mono") { machine = MCH_CGA; mono_cga = true; }
-	else if (mtype == "tandy")    { machine = MCH_TANDY; }
-	else if (mtype == "pcjr")     { machine = MCH_PCJR;
-	} else if (mtype == "hercules") {
-		machine = MCH_HERC;
-	} else if (mtype == "ega") {
-		machine = MCH_EGA;
-	} else if (mtype == "svga_s3") {
-		svgaCard = SVGA_S3Trio;
-	} else if (mtype == "vesa_nolfb") {
-		svgaCard = SVGA_S3Trio;
+
+	if (machine_str == "cga") {
+		machine = MachineType::Cga;
```

Comment:
`MachineType::CgaMono` and dump the flag? It's a nasty hack.

Reply:
Done. Also renamed `MachineType::Cga` to `MachineType::CgaColor`, otherwise `is_machine_cga()` name was a little ambiguous.